neptune/management/exceptions.py

```diff
@@ -85,14 +85,15 @@ class UserNotExistsOrWithoutAccess(ManagementOperationFailure):
     code = 9
     description = (
         'User "{user}" does not exist or has no access to project "{project}". '
-        "If the project visibility is set to workspace, a user cannot be added or removed."
+        "If the project visibility is set to 'workspace', a user cannot be added or removed."
```

Comment:
Is the wording "added or removed" correct? As far as I can tell, this exception is raised only when you try to remove a user from a project.

Could you confirm whether the removal will fail in the following scenarios:

1. The user was not found among the members of a private project
1. The project's visibility is "workspace" or "public"

Reply:
User removal in all visibility scenario will raise `UserNotExistsOrWithoutAccess`
In case of adding a user to the project, it will pass in with all visibilities (in case of "workspace" visibility it will not be listed after all but will be stored internally and will be visible after switching to private visibility).

Reply:
Currently adding users to workspace visibility is some kind of unexpected behavior that we've introduced due to easier implementation. Could be treated as a bug.

Reply:
If I understand your reply correctly, this exception would not be raised when you add a user to a project with `visibility="workspace"`, correct? Just trying to understand if the highlighted line is accurate or not :)

Reply:
Yes, It won't be raised (currently, probably this will be updated in the future)

Reply:
Okay. There are some other potential clarifications that could be made to project visibility documentation, so I'm going to resolve this in the context of this PR.